src/main/java/me/alexpresso/zuninja/classes/projection/ItemProjection.java

```diff
@@ -2,21 +2,60 @@
 
 import me.alexpresso.zuninja.classes.projection.action.ActionElement;
 import me.alexpresso.zuninja.domain.nodes.item.Item;
+import me.alexpresso.zuninja.domain.relations.InputToFusion;
 import me.alexpresso.zuninja.domain.relations.InventoryItem;
 
+import java.util.Optional;
+
 public class ItemProjection implements ActionElement {
     private final Item item;
     private int quantity;
     private int upgradeLevel;
+    private final InventoryProjection inventory;
+    private final boolean golden;
+    private ItemCountProjection countProjection;
 
 
-    public ItemProjection(final InventoryItem iv) {
-        this(iv.getItem(), iv.getQuantity(), iv.getUpgradeLevel());
+    public ItemProjection(final InventoryItem iv, final InventoryProjection inventory, final boolean golden) {
+        this(iv.getItem(), iv.getQuantity(), iv.getUpgradeLevel(), inventory, golden);
     }
-    public ItemProjection(final Item item, final int quantity, final int upgradeLevel) {
+    public ItemProjection(final Item item,
+                          final int quantity,
+                          final int upgradeLevel,
+                          final InventoryProjection inventory,
+                          final boolean golden) {
         this.item = item;
         this.quantity = quantity;
         this.upgradeLevel = upgradeLevel;
+        this.inventory = inventory;
+        this.golden = golden;
+
+        this.initCountProjection();
```

Comment:
Wait for every items to be filled in inventories before init (getQuantity can be wrong if some items aren't filled yet)